File: bindings/node/src/detect-libc.js
'use strict';

const libcFromLdd = require('./libc-from-ldd');
const libcFromReport = require('./libc-from-report');

// 'glibc', 'musl' or null: /usr/bin/ldd first, then the diagnostic report.
// The bridge checks the libc itself because an older npm ignores the libc
// field of the platform packages.
function detectLibc(probe) {
  return libcFromLdd(probe.ldd()) || libcFromReport(probe.report());
}

module.exports = detectLibc;

File: bindings/go/detect_libc.go
package memless

const lddPath = "/usr/bin/ldd"

// detectLibc returns "glibc", "musl" or "": /usr/bin/ldd first, then the
// dynamic loader under /lib, for images that ship no ldd.
func detectLibc(readFile func(string) ([]byte, error), exists func(string) bool) string {
	data, err := readFile(lddPath)
	if libc := libcFromLdd(data, err); libc != "" {
		return libc
	}
	return libcFromLoaders(exists)
}
